test/extended/router/scoped.go

```diff
@@ -241,24 +241,31 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 func waitForRouterOKResponseExec(ns, execPodName, url, host string, timeoutSeconds int) error {
 	cmd := fmt.Sprintf(`
 		set -e
-		STOP=$(($(date '+%%s') + %d))
+		pass=%[4]d
+		STOP=$(($(date '+%%s') + %[1]d))
 		while [ $(date '+%%s') -lt $STOP ]; do
 			rc=0
-			code=$( curl -k -s -m 5 -o /dev/null -w '%%{http_code}\n' --header 'Host: %s' %q ) || rc=$?
+			code=$( curl -k -s -m 5 -o /dev/null -w '%%{http_code}\n' --header 'Host: %[2]s' %[3]q ) || rc=$?
 			if [[ "${rc:-0}" -eq 0 ]]; then
 				echo $code
 				if [[ $code -eq 200 ]]; then
-					exit 0
+					pass=$(( pass - 1 ))
+					if [[ $pass -lt 0 ]]; then
```

Comment:
Off by one?  If `pass` starts at 5, this logic curls the route 6 times, which is unintuitive.

Reply:
Yeah, will change to `-le`.